lambda_code/src/fhir_controller.py

```diff
@@ -66,6 +63,19 @@ def delete_immunization(self, aws_event):
         except UnhandledResponseError as unhandled_error:
             return self.create_response(500, unhandled_error.to_operation_outcome().json())
 
+    def search_immunizations(self, aws_event) -> dict:
+        params = aws_event["queryStringParameters"]
+        print(aws_event)
+
+        if "nhsNumber" not in params:
```

Comment:
Should check that the value isn't None or whitespace.

Reply:
This part should be done when validation code is wired in. For now I just assumed the object is valid.